src/Courses/TopicGrades/GradingPage.tsx

```diff
@@ -0,0 +1,259 @@
+        path?: string,
+        seed?: number,
+        grade?: StudentGradeDict,
+        problem?: ProblemObject,
+        workbooks?: Record<number, StudentWorkbookInterface>,
+        workbook?: StudentWorkbookInterface,
+    }>({});
+
+    useEffect(() => {
+        logger.debug('GP2: topicId changed');
+        (async () => {
+            try {
+                if (_.isNil(params.topicId)) {
+                    logger.error('topicId is null');
+                    throw new Error('An unexpected error has occurred');
+                } else {
+                    await fetchProblems(parseInt(params.topicId, 10));
+                }
+            } catch (e) {
+                logger.error(e.message, e);
+            }
+        })();
+    }, [params.topicId]);
+
+    useEffect(() => {
+        logger.debug('GP2: different user or problem was selected');
+        // when user and problem are selected - set the available workbooks and 
+        // pick one workbook as the default for rendering
+        // TODO: adjust for different policies -- best individual / best attempt
+        let currentPath: string | undefined;
+        let currentSeed: number | undefined;
+        let currentUserGrade: StudentGradeDict | undefined;
+        let currentWorkbooks: Record<number, StudentWorkbookInterface> | undefined;
+        let currentWorkbook: StudentWorkbookInterface | undefined;
+        
+        if (!_.isNil(selected.problem) && !_.isNil(selected.user)) {
+            if (!_.isNil(selected.problem.webworkQuestionPath)) {
+                currentPath = selected.problem.webworkQuestionPath;
+            } 
+            if (!_.isNil(selected.problem.id) && !_.isNil(selected.user.id)) {
+                const problemDict = problemMap[selected.problem.id];
+                if (!_.isNil(problemDict.grades)) {
+                    currentUserGrade = problemDict.grades[selected.user.id];
+                    if (!_.isNil(currentUserGrade.randomSeed)) {
+                        currentSeed = currentUserGrade.randomSeed;
+                    }
+                    const userProblemWorkbooks = currentUserGrade.workbooks;
+                    const selectedWorkbookId = currentUserGrade.lastInfluencingAttemptId; // this can be expanded
+                    if (!_.isNil(userProblemWorkbooks) && (!_.isEmpty(userProblemWorkbooks))) {
+                        currentWorkbooks = userProblemWorkbooks;
+                        if (!_.isNil(selectedWorkbookId)) {
+                            currentWorkbook = userProblemWorkbooks[selectedWorkbookId];
+                            if (_.isNil(currentWorkbook)) {
+                                logger.error(`we were supposed to get workbook #${selectedWorkbookId}, but failed.`);
+                            } else {
+                                // logger.debug(currentWorkbook.id);
+                                currentPath = undefined;
+                                currentSeed = undefined;
```

Comment:
Why do these get unset if there's a currentWorkbook?

Reply:
to prevent the iframe from loading the preview path / preview seed